CorsixTH/Lua/dialogs/tree_ctrl.lua

```diff
@@ -604,6 +604,11 @@ function TreeControl:setSelectCallback(callback)
   return self
 end
 
+function TreeControl:setTextboxCallback(callback)
```

Comment:
Needs luadoc

Also, the name is **very** implementation oriented (`updateTextBox` has that problem too).

That is, if I wanted to change a label instead of a textbox, or I wanted to write some data to a remote UI, the name doesn't make much sense any more while it's purpose is still valid. This is a sign that the abstraction level in the name is too low.

The way out of this is take a step back, and ask yourself: What is the purpose of this call, why do you make this call at all? Use that as input for finding a name.  
Such higher level goals in software are much more stable than current code patterns.

Higher level names also makes it easier to connect code to such higher level goals, which usually improves understanding.

Reply:
In all fairness, giving good names is hard to get right. It needs lots of practice and even then we make a mess sometimes.

Reply:
Maybe `TreeControl:changeValueCallback(callback)`?